tangelo/toolboxes/post_processing/histogram.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2021 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" This module provides a Histogram class and functions to manipulate them,
+in order to facilitate post-processing of quantum experiments.
+"""
+
+from collections import Counter
+
+from tangelo.linq import Simulator
+from tangelo.toolboxes.post_processing.bootstrapping import get_resampled_frequencies
+
+
+class Histogram:
+    """Class to provide useful tools helping redundant tasks when analyzing
+    data from an experiment. The expected data input is an histogram of
+    bistrings ("010..."), where 0/1 correspond to the |0>/|1> measured state in
+    the computational basis. The outcomes can refer to either the number of
+    replicas (where the summation across the bistrings equals the number of
+    shots), or the probability (where the summation equals to 1). In the former
+    case, the number of shots must be provided.
+
+    The internal representation is kept as an histogram of shots. Normalized
+    quantities are accessible via the Histogram "frequencies" properties.
+
+    Args:
+        outcomes (dict): Results in the format of bistring: outcome, where
+            outcome can be an int (number of replicas) or a float (probability).
+        n_shots (int): Self-explanatory. If it is greater than 0, the class
+            considers that probabilities are provided.
+        msq_first (bool): Bit ordering. For example, 011 (msq_first) = 110
+            (lsq_first). This depends on the hardware provider convention.
+        epsilon (float): When probabilities are provided, this parameter is used
+            to check if the sum is within [1-epsilon, 1+epsilon]. Default is
+            1e-2.
+    """
+
+    def __init__(self, outcomes, n_shots=0, msq_first=False, epsilon=1e-2):
```

Comment:
What makes more sense to you: default value for shots=0 or `None` ?

Regardless of what you pick, a number of operations here will return strange things if n_shots is not a strictly positive number, we may want to to account for that with a warning message, errors, or something like that.